internal/servers/controller/handlers/authmethods/authmethod_service.go

```diff
@@ -128,14 +128,26 @@ func (s Service) ListAuthMethods(ctx context.Context, req *pbs.ListAuthMethodsRe
 	}
 	authResults := s.authResult(ctx, req.GetScopeId(), action.List)
 	if authResults.Error != nil {
-		return nil, authResults.Error
+		// If it's forbidden, and it's a recursive request, and they're
+		// successfully authenticated but just not authorized, keep going as we
+		// may have authorization on downstream scopes.
+		if authResults.Error == handlers.ForbiddenError() &&
```

Comment:
is there a anyway to provide a func that combines authResults(...) and GetScopeIds(...) that could be used in all the List functions?  It seems like this block is going to be copied quite a bit.

Reply:
I'll do that as a follow after this PR. I wanted to keep changes minimal (as opposed to additions) for easy reviewing.